packages/frontend/src/hooks/echarts/useEchartsMapConfig.ts

```diff
@@ -0,0 +1,588 @@
+
+                        const lat = Number(row[latitudeFieldId]?.value.raw);
+                        const lon = Number(row[longitudeFieldId]?.value.raw);
+                        const value = valueFieldId
+                            ? Number(row[valueFieldId]?.value.raw)
+                            : 1;
+
+                        if (isNaN(lat) || isNaN(lon)) return null;
+
+                        return {
+                            value: [lon, lat, value] as [
+                                number,
+                                number,
+                                number,
+                            ],
+                            rowData: row,
+                        };
+                    })
+                    .filter((d): d is NonNullable<typeof d> => d !== null);
+            } else {
+                // Region/Country choropleth map
+                regionData = resultsData.rows
+                    .map((row) => {
+                        if (!locationFieldId) return null;
+
+                        const locationName = String(
+                            row[locationFieldId]?.value.raw || '',
+                        );
+                        const value = valueFieldId
+                            ? Number(row[valueFieldId]?.value.raw)
+                            : 1;
+
+                        if (!locationName) return null;
+
+                        return {
+                            name: locationName,
+                            value: isNaN(value) ? 0 : value,
+                        };
+                    })
+                    .filter((d): d is NonNullable<typeof d> => d !== null);
+            }
+        }
+
+        // Create d3 scale for scatter point sizing
+        const scatterValues = scatterData.map((d) => d.value[2]);
+        const bubbleSizeMin = minBubbleSize ?? 5;
+        const bubbleSizeMax = maxBubbleSize ?? 20;
+        const sizeScale =
+            scatterValues.length > 0
+                ? scaleSqrt()
+                      .domain([
+                          Math.min(...scatterValues, 0),
+                          Math.max(...scatterValues, 1),
+                      ])
+                      .range([bubbleSizeMin, bubbleSizeMax])
+                : () => bubbleSizeMin;
+
+        // Check if this is a pre-projected map (Albers) that needs identity projection
+        const isPreProjected =
+            mapType === MapChartLocation.USA ||
+            mapType === MapChartLocation.USA_COUNTIES;
+        const projection = isPreProjected
+            ? {
+                  project: (point: [number, number]) => point,
+                  unproject: (point: [number, number]) => point,
+              }
+            : undefined;
+
+        // Clamp zoom to valid range (1-10)
+        const clampedZoom = defaultZoom
+            ? Math.min(10, Math.max(1, defaultZoom))
+            : undefined;
+
+        if (isLatLong) {
```

Comment:
depending on the map type, the echarts config is different, perhaps we can extract this to aseparate function